popup/search-results.js

```diff
@@ -0,0 +1,547 @@
+    const description = $('.searchResult-description', entry);
+    Object.assign(description, {
+      textContent: userstyleSearchResult.description.replace(/<.*?>/g, ''),
+      title: userstyleSearchResult.description.replace(/<.*?>/g, '')
+    });
+
+    const authorLink = $('.searchResult-authorLink', entry);
+    Object.assign(authorLink, {
+      textContent: userstyleSearchResult.user.name,
+      title: userstyleSearchResult.user.name,
+      href: 'https://userstyles.org/users/' + userstyleSearchResult.user.id,
+      onclick: handleEvent.openURLandHide
+    });
+
+    const rating = $('.searchResult-rating', entry);
+    let ratingClass;
+    let ratingValue = userstyleSearchResult.rating;
+    if (ratingValue === null) {
+      ratingClass = 'none';
+      ratingValue = 'n/a';
+    } else if (ratingValue >= 2.5) {
+      ratingClass = 'good';
+      ratingValue = ratingValue.toFixed(1);
+    } else if (ratingValue >= 1.5) {
+      ratingClass = 'okay';
+      ratingValue = ratingValue.toFixed(1);
+    } else {
+      ratingClass = 'bad';
+      ratingValue = ratingValue.toFixed(1);
+    }
+    Object.assign(rating, {
+      textContent: ratingValue,
+      className: 'searchResult-rating ' + ratingClass
+    });
+
+    const installCount = $('.searchResult-installCount', entry);
+    Object.assign(installCount, {
+      textContent: userstyleSearchResult.total_install_count.toLocaleString()
+    });
+
+    const installButton = $('.searchResult-install', entry);
+    installButton.onclick = install;
+
+    if (userstyleSearchResult.style_settings.length > 0) {
+      // Style has customizations
+      installButton.classList.add('customize');
+      const customizeButton = $('.searchResult-customize', entry);
+      customizeButton.classList.remove('hidden');
+      customizeButton.href = 'https://userstyles.org' + userstyleSearchResult.url;
+      customizeButton.onclick = handleEvent.openURLandHide;
+    }
+
+    /** Installs the current userstyleSearchResult into stylus. */
+    function install() {
+      entry.classList.add('loading');
+
+      const styleId = userstyleSearchResult.id;
+      const url = 'https://userstyles.org/styles/chrome/' + styleId + '.json';
+      saveStyleSafe(userstyleSearchResult.json)
+        .then(() => {
+          // Remove search result after installing
+          let matchingIndex = -1;
+          processedResults.forEach((processedResult, index) => {
+            if (processedResult.id === userstyleSearchResult.id) {
+              matchingIndex = index;
+            }
+          });
+          if (matchingIndex >= 0) {
+            processedResults.splice(matchingIndex, 1);
+          }
+          processNextResult();
+        })
+        .catch(reason => {
+          console.log('install:saveStyleSafe(', url, ') => [ERROR]: ', reason);
+          alert('Error while downloading ' + url + '\nReason: ' + reason);
+        });
+      return true;
+    }
+  }
+})();
+
+onDOMready().then(() => {
```

Comment:
Move it to the top of the file so it's obvious what happens on including the js right away